lib/pbench/test/functional/server/test_put.py

```diff
@@ -0,0 +1,63 @@
+from pathlib import Path
+import time
+
+from pbench.client import PbenchServerClient
+
+TARBALL_DIR = Path("lib/pbench/test/functional/server/tarballs")
+
+
+class TestPut:
+    tarballs: list[Path] = []
+
+    def test_upload_all(self, server_client: PbenchServerClient, login_user):
+        """Upload each of the pregenerated tarballs, and perform some basic
+        sanity checks on the resulting server state.
+        """
+
+        for t in TARBALL_DIR.glob("*.tar.xz"):
+            self.tarballs.append(t)
+            response = server_client.upload(t)
+            assert response.ok
+        datasets = server_client.get_list(
+            metadata=["server.tarball-path", "server.status"]
+        )
+        found = sorted(d.name for d in datasets)
+        expected = sorted(t.name[:-7] for t in self.tarballs)
```

Comment:
Perhaps,
```
        expected = sorted(Dataset.stem(t.name) for t in self.tarballs)
```
(The magic of `[:-7]` is everywhere!)

Reply:
Some of this is older code; I did add a `Dataset.stem` which could be retrofitted to this code.